### PR TITLE
Memory game

### PR DESCRIPTION
**Pull request by madmaxWMFU**
### Memory – Pair Game

[Demo](https://madmaxwmfu.github.io/memory-game/) |
[Code base](https://github.com/madmaxWMFU/memory-game)

Please, review.

### CODE REVIEW

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const main = document.querySelector(".game-field");
+const arrayOfCards = ["card0", "card1", "card2", "card3", "card4", "card5", "card6", "card7", "card8", "card9", "card10", "card11", "card12", "card13", "card14", "card15", "card16", "card17", "card18", "card19", "card20", "card21", "card22", "card23", "card24", "card25", "card26", "card27", "card28", "card29", "card30", "card31", "card32", "card33", "card34", "card35", "card36", "card37", "card38", "card39", "card40", "card41", "card42", "card43", "card44", "card45", "card46", "card47", "card48", "card49", "card50", "card51", "card52", "card53"];
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: This is overcomplicated defined array. Can we use some method for generate them?

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+let twoSelectedCads = [];
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: twoSelectedCa **r** ds
What happens if user selected one of cards. `twoSelectedCards` will be with one values ? IMO better name will be just `selectedCards`.

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const shuffled = (arr) => arr.sort(() => 0.5 - Math.random());
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: `shuffleCards` name give us more understanding what happens inside method.

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+const drawCards = arrayOfCards => {
+	let selected = shuffled(arrayOfCards).slice(0, 10);
+	let playCards = shuffled([...selected, ...selected]);
+	playCards.forEach(val => main.insertAdjacentHTML("beforeEnd", getTemplateWrap(val)));
+}
+
+const getTemplateWrap = (newClass) => {
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: Please, use one - style for defining one properties with arrow function. It's up to you, with braces or without.

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    return `
+        <div class="flipper hideCard">
+            <div class="front"></div>
+            <div class="back ${newClass}" data-value="${newClass}"></div>
+        </div>`;
+}
+
+const hideAllSelectedCards = () => document.querySelectorAll(".flipper").forEach(item => hideCard(item));
+const hideCheckedCards = () => document.querySelectorAll(".rotateCard").forEach(item => item.style.visibility = "hidden");
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: This is a bad practice styling elements by javascript. Use classes for that.

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+const hideCard = item => {
+	item.classList.remove("rotateCard");
+	item.classList.add("hideCard");
+}
+
+const rotateCard = (item) => {
+	item.classList.remove("hideCard");
+	item.classList.add("rotateCard");
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: Can you try use `toggle` method of `classList` ?

File: submissions/madmaxWMFU/memory-game/style/style.css [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+    transform: rotateY(180deg); 
+}
+
+.hideCard {    
+    transform: rotateY(0deg);
+}
+
+.card0 {
+    background: url('../img/koloda.png') no-repeat 0 0;
+    background-size: 2529px 947px;
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/style/style.css]: Can we define common properties for all cards in another css class ? `background-size` repeated.

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const countSuccessfullCard = () => {
+	if(document.querySelectorAll("[style ^='visibility']").length == 20) {
+    	alert("Congratulations! You found all the cards.");
+    	init();
```
Comment (o-msh) [inline submissions/madmaxWMFU/memory-game/js/app.js]: Please, use more prettier code with correct spaces.

**Comment by madmaxWMFU** [comment]
@o-msh thanks for your feedback!
I corrected all the shortcomings

File: submissions/madmaxWMFU/memory-game/style/style.css (hunk 7 above)
Comment (madmaxWMFU) [reply submissions/madmaxWMFU/memory-game/style/style.css]: I rewrote the rule, but I can not remove the property `background-size`, without it the style does not work

File: submissions/madmaxWMFU/memory-game/js/app.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+		});
+		clearOpenCards();
+		countSuccessfullCard();
+	}, hideDelay);
+}
+
+const showCard = (item) => item.classList.toggle("rotate-card");
+
+const countSuccessfullCard = () => {
+	if(document.querySelectorAll(".remove-card").length == 20) {
```
[the remark this answers is not in the document]
Comment (o-msh) [reply submissions/madmaxWMFU/memory-game/js/app.js]: Also better use a strong checking. `===`

**Comment by o-msh** [comment]
@madmaxWMFU I'll merge your PR. IMO you're really improved code and knowing about some new feautures. LGTM!

**Comment by madmaxWMFU** [comment]
@o-msh thanks a lot !!

**Pull request merged**